SeamCarver.js

```diff
@@ -203,13 +203,38 @@ class SeamCarver {
     }
 
     /**
+     * Removes vertical seam.
+     * Recalculates pixels depending on removed pixel.
+     *
+     */
+    removeVerticalSeam(vseam) {
+        for (var row = 0; row < vseam.length; row ++) {
+            var deletedCol = vseam[row];
+
+            // Can ignore last column as we will delete it
+            for (var col = 0; col < this.width - 1; col ++) {
+                if (col >= deletedCol) {
+                    this.energy_matrix[col][row] = this.energy_matrix[col + 1][row];
```

Comment:
It seems to me this is a very inefficient way of traversing the matrix. If we wish to benefit from CPU caching we should either arrange the data structure into an array of rows or iterate each row for each col.